tests/framework/e2e/etcd_process.go

```diff
@@ -72,6 +73,8 @@ type EtcdServerProcess struct {
 	proxy      proxy.Server
 	failpoints *BinaryFailpoints
 	donec      chan struct{} // closed when Interact() terminates
+
+	keptLogLines []string // only filled when KeepLogs=true
```

Comment:
Not sure why this is needed. ExpectLogs already stores all logs in memory. Don't see a problem to always keep logs.

Reply:
it's because of the ep.proc lifecycle below:

```
		if ep.cfg.KeepLogs {
			ep.keptLogLines = ep.proc.Lines()
		}
***		ep.proc = nil ***
```

it's also used to flag whether a process was stopped, which coincidentally wipes the logs with it.

Reply:
Then I would suggest storing `ExpectLog` interface instead of `[]string{}`

Reply:
sounds good